### PR TITLE
[Merged by Bors] - feat: `1` is the only positive element of finite order

### PR DESCRIPTION
**Pull request by YaelDillies**
and other simple lemmas about the order of elements in a group

From LeanAPAP

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/GroupTheory/OrderOfElement.lean [hunk 1]
```diff
@@ -1040,18 +1045,20 @@ theorem Subgroup.pow_index_mem {G : Type*} [Group G] (H : Subgroup G) [Normal H]
...
-#align pow_eq_mod_card pow_eq_mod_card
-#align nsmul_eq_mod_card nsmul_eq_mod_card
 
-@[to_additive]
-theorem zpow_eq_mod_card (n : ℤ) : x ^ n = x ^ (n % Fintype.card G : ℤ) := by
-  rw [← zpow_mod_orderOf, ← Int.emod_emod_of_dvd n (Int.coe_nat_dvd.2 orderOf_dvd_card),
+@[to_additive (attr := simp) mod_card_nsmul]
+lemma pow_mod_card (a : G) (n : ℕ) : a ^ (n % card G) = a ^ n :=
+  mul_left_injective (a ^ (card G * (n / card G))) $ by
+    simp_rw [←pow_add, Nat.mod_add_div, pow_add, pow_mul, pow_card_eq_one, one_pow, mul_one]
```
Comment (sgouezel) [inline Mathlib/GroupTheory/OrderOfElement.lean:1052]: the previous proof was simpler, why did you change it?

File: Mathlib/GroupTheory/OrderOfElement.lean [hunk 2]
```diff
@@ -1164,9 +1175,23 @@ def powCardSubgroup {G : Type*} [Group G] [Fintype G] (S : Set G) (hS : S.Nonemp
 
 end PowIsSubgroup
 
+section LinearOrderedSemiring
+variable [LinearOrderedSemiring G] {a : G}
+
+protected lemma IsOfFinOrder.eq_one (ha₀ : 0 ≤ a) : IsOfFinOrder a → a = 1 := by
```
Comment (sgouezel) [inline Mathlib/GroupTheory/OrderOfElement.lean:1181]:
```suggestion
protected lemma IsOfFinOrder.eq_one (ha₀ : 0 ≤ a) (ha : IsOfFinOrder a) : a = 1 := by
```
It's easier to read that way, even if it needs a few tweaks in the proof.

**Review by sgouezel — COMMENTED** [review]
bors d+

File: Mathlib/GroupTheory/OrderOfElement.lean (hunk 1 above)
Comment (YaelDillies) [reply Mathlib/GroupTheory/OrderOfElement.lean:1052]: The `rw` now targets the wrong `^`, but I agree I made it too complicated.

Comment (sgouezel) [reply Mathlib/GroupTheory/OrderOfElement.lean:1052]: You can just start with `rw [eq_comm]`...

**Comment by YaelDillies** [comment]
bors merge

**Comment by YaelDillies** [comment]
bors merge
